app/infrastructure/db/repositories/in_memory_user_repository.py

```
@@ -54,3 +54,21 @@ def update_password(self, user_id: str, password_hash: str) -> User | None:
         return {uid: user for uid, user in self._by_id.items() if uid in user_ids}
+
+    def get_by_provisioning_token(self, token: str) -> User | None:
+        for user in self._by_id.values():
+            if user.provisioning_token == token:
+                return user
+        return None
+
+    def set_provisioning_token(self, user_id: str, token: str) -> None:
+        user = self._by_id.get(user_id)
+        if user is None:
+            raise ValueError(f"User {user_id} not found")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)